testing/e2e/clients/aks.go

```diff
@@ -119,43 +128,160 @@ func (a *aks) Deploy(ctx context.Context, objs []client.Object) error {
-	encoded := base64.StdEncoding.EncodeToString(b.Bytes())
+	return b.Bytes(), nil
+}
 
-	poller, err := client.BeginRunCommand(ctx, a.resourceGroup, a.name, armcontainerservice.RunCommandRequest{
-		Command: to.Ptr("kubectl apply -f manifests/"),
+func (a *aks) Clean(ctx context.Context, objs []client.Object) error {
+	lgr := logger.FromContext(ctx).With("name", a.name, "resourceGroup", a.resourceGroup)
+	ctx = logger.WithContext(ctx, lgr)
+	lgr.Info("starting to clean resources")
+	defer lgr.Info("finished cleaning resources")
+
+	zip, err := zipManifests(objs)
+	if err != nil {
+		return fmt.Errorf("zipping manifests: %w", err)
+	}
+	encoded := base64.StdEncoding.EncodeToString(zip)
+
+	if err := a.runCommand(ctx, armcontainerservice.RunCommandRequest{
+		Command: to.Ptr("kubectl delete -f manifests/"),
 		Context: &encoded,
-	}, nil)
+	}, runCommandOpts{}); err != nil {
+		return fmt.Errorf("running kubectl delete: %w", err)
+	}
+
+	return nil
+}
+
+func (a *aks) waitStable(ctx context.Context, objs []client.Object) error {
+	lgr := logger.FromContext(ctx).With("name", a.name, "resourceGroup", a.resourceGroup)
+	ctx = logger.WithContext(ctx, lgr)
+	lgr.Info("starting to wait for resources to be stable")
+	defer lgr.Info("finished waiting for resources to be stable")
+
+	var eg errgroup.Group
+	for _, obj := range objs {
+		func(obj client.Object) {
+			eg.Go(func() error {
+				kind := obj.GetObjectKind().GroupVersionKind().GroupKind().Kind
+				ns := obj.GetNamespace()
+				if ns == "" {
+					ns = "default"
+				}
+
+				lgr := lgr.With("kind", kind, "name", obj.GetName(), "namespace", ns)
+				lgr.Info("checking stability of " + kind + "/" + obj.GetName())
+
+				switch {
+				case slices.Contains(workloadKinds, kind):
+					lgr.Info("checking rollout status")
+					if err := a.runCommand(ctx, armcontainerservice.RunCommandRequest{
+						Command: to.Ptr(fmt.Sprintf("kubectl rollout status %s/%s -n %s", kind, obj.GetName(), ns)),
+					}, runCommandOpts{}); err != nil {
+						return fmt.Errorf("waiting for %s/%s to be stable: %w", kind, obj.GetName(), err)
+					}
+				case kind == "Pod":
+					lgr.Info("waiting for pod to be ready")
+					if err := a.runCommand(ctx, armcontainerservice.RunCommandRequest{
+						Command: to.Ptr(fmt.Sprintf("kubectl wait --for=condition=Ready pod/%s -n %s", obj.GetName(), ns)),
+					}, runCommandOpts{}); err != nil {
+						return fmt.Errorf("waiting for pod/%s to be stable: %w", obj.GetName(), err)
+					}
+				case kind == "Job":
+					lgr.Info("waiting for job complete")
+					if err := a.runCommand(ctx, armcontainerservice.RunCommandRequest{
+						Command: to.Ptr(fmt.Sprintf("kubectl logs --pod-running-timeout=20s --follow job/%s -n %s", obj.GetName(), ns)),
+					}, runCommandOpts{
+						outputFile: fmt.Sprintf("job-%s.log", obj.GetName()), // output to a file for jobs because jobs are naturally different from other deployment resources in that waiting for "stability" is waiting for them to complete
```

Comment:
simply because jobs can be annoying about not being able to redeploy them without deleting the original one - do we want to include cleanup here once we know the job is done/once we've gotten its output? or just leave that to user/test dev?

Reply:
this would only really be useful in the event of a failed job/test so not sure if it's necessary/desirable to cater to that scenario

Reply:
We have a cleanup step already it's just not bundled with the stability checker since stability has a different goal. We attempt to cleanup any old jobs before deploying a new one.

Reply:
do you mean in deploy_e2e_tester.sh? will we still use that script when we transition to this go-based provisioning?